### ISSUE TITLE
Remove "warning" that assembly has already been processed

### ISSUE THREAD

**Issue by drauch** [Fody/Fody#1023]
#### Describe the issue

If you build a project twice, the second time you receive build warnings:

> MSBUILD : warning : Fody: The assembly has already been processed by Fody. Weaving aborted.

This is a problem for everybody who uses "warnings as errors", because it results in a failed build if you do not clean everything beforehand.

#### Make an effort to fix the bug

Convert the log statement in https://github.com/Fody/Fody/blob/96d8ebb4a4581191760e1758a1103fe83c8e9f17/FodyIsolated/InnerWeaver.cs#L103 to an informational level log message.

Best regards,
D.R.

**Comment by SimonCropp** [comment]
just to confirm you are a patron on open collective?

**Comment by drauch** [comment]
No I'm in fact not (yet). I still bring problems to your attention, it does not hurt you, and it helps others who run into the same problems.

Meta:
If you decide to not fix this problem because I'm not a sponsor (yet), I can certainly understand the motives, although, in case of problem reports I can't see that the decision to not address problems is helping the product. Honestly, I thought the policy was something you intended for feature requests and new ideas, especially after reading:
> Issues from non-Patrons, that are bugs and are highly likely to effect the larger Fody user base, may justify the effort and be handled.

**Comment by SimonCropp** [comment]
can u push a minimal solution that reproduces this issue to a github repo

**Comment by tranb3r** [comment]
Hi,
Here is a simple [repro](https://github.com/tranb3r/Issues/tree/main/MauiAppFodyAlreadyProcessed) for a similar issue, happening in a Maui application.
To reproduce, checkout the repro, then go to `scripts` folder, and run `clean.bat` then `publish_android.bat`.
Here is the build log with the warning:
```
MSBuild version 17.4.0+18d5aef85 for .NET
MSBuild version 17.4.0+18d5aef85 for .NET
C:\Program Files\dotnet\sdk\7.0.100\MSBuild.dll -bl -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\7.0.100\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\7.0.100\dotnet.dll -maxcpucount -property:Configuration=Release -property:RuntimeIdentifier=android-arm64 -property:_CommandLineDefinedRuntimeIdentifier=true -property:SelfContained=True -property:_CommandLineDefinedSelfContained=true -target:Restore -verbosity:m -verbosity:m ../MauiAppFodyAlreadyProcessed.Maui/MauiAppFodyAlreadyProcessed.Maui.csproj
  Determining projects to restore...
  Restored D:\Dev\Issues\MauiAppFodyAlreadyProcessed\MauiAppFodyAlreadyProcessed.Lib\MauiAppFodyAlreadyProcessed.Lib.csproj (in 416 ms).
  Restored D:\Dev\Issues\MauiAppFodyAlreadyProcessed\MauiAppFodyAlreadyProcessed.Maui\MauiAppFodyAlreadyProcessed.Maui.csproj (in 416 ms).
C:\Program Files\dotnet\sdk\7.0.100\MSBuild.dll -nologo -bl -consoleloggerparameters:Summary -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\7.0.100\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\7.0.100\dotnet.dll -maxcpucount -property:TargetFramework=net7.0-android33.0 -property:Configuration=Release -property:RuntimeIdentifier=android-arm64 -property:_CommandLineDefinedRuntimeIdentifier=true -property:SelfContained=True -property:_CommandLineDefinedSelfContained=true -verbosity:m -verbosity:m ../MauiAppFodyAlreadyProcessed.Maui/MauiAppFodyAlreadyProcessed.Maui.csproj
  MauiAppFodyAlreadyProcessed.Lib -> D:\Dev\Issues\MauiAppFodyAlreadyProcessed\MauiAppFodyAlreadyProcessed.Lib\bin\Release\net7.0-android33.0\MauiAppFodyAlreadyProcessed.Lib.dll
  MauiAppFodyAlreadyProcessed.Maui -> D:\Dev\Issues\MauiAppFodyAlreadyProcessed\MauiAppFodyAlreadyProcessed.Maui\bin\Release\net7.0-android33.0\android-arm64\MauiAppFodyAlreadyProcessed.Maui.dll
MSBUILD : warning : Fody: The assembly has already been processed by Fody. Weaving aborted. Path: D:\Dev\Issues\MauiAppFodyAlreadyProcessed\MauiAppFodyAlreadyProcessed.Lib\obj\Release\net7.0-android33.0\MauiAppFodyAlreadyProcessed.Lib.dll [D:\Dev\Issues\MauiAppFodyAlreadyProcessed\MauiAppFodyAlreadyProcessed.Lib\MauiAppFodyAlreadyProcessed.Lib.csproj::TargetFramewo
rk=net7.0-android33.0]
  MauiAppFodyAlreadyProcessed.Lib -> D:\Dev\Issues\MauiAppFodyAlreadyProcessed\MauiAppFodyAlreadyProcessed.Lib\bin\Release\net7.0-android33.0\android-arm64\MauiAppFodyAlreadyProcessed.Lib.dll
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Optimizing assemblies for size. This process might take a while.
```
Could you please reopen this issue ? Or maybe I should open a new ticket ?
Thanks

**Comment by tom-englert** [comment]
@tranb3r can you set your log level to detailed to see where Fody is called twice?

**Comment by tranb3r** [comment]
msbuild detailed log (text and binary):
[msbuildlog.zip](https://github.com/Fody/Fody/files/10219895/msbuildlog.zip)

**Comment by tranb3r** [comment]
@tom-englert 
Any update? Thanks

**Comment by tom-englert** [comment]
@tranb3r sorry, lost track of this.
Can you try adding this property to your project:
`<DisableFody Condition="'_ComputeFilesToPublishForRuntimeIdentifiers'=='true'">true</DisableFody>`

**Comment by tranb3r** [comment]
> @tranb3r sorry, lost track of this. Can you try adding this property to your project: `<DisableFody Condition="'_ComputeFilesToPublishForRuntimeIdentifiers'=='true'">true</DisableFody>`

@tom-englert 
It doesn't make any difference.

**Comment by tom-englert** [comment]
@tranb3r can you post the binary log when you have this setting on?

**Comment by tranb3r** [comment]
@tom-englert 
Here is the bin log with the setting on (in both projects)
[msbuild_test.zip](https://github.com/Fody/Fody/files/10534316/msbuild_test.zip)

**Comment by tom-englert** [comment]
can you try this instead:
```xml
<Target Name="DisableFodyOnComputeFiles" BeforeTargets="CoreCompile" Condition="'_ComputeFilesToPublishForRuntimeIdentifiers'=='true'">
    <PropertyGroup>
        <DisableFody>true</DisableFody>
    </PropertyGroup>
    <Message Text="DisableFodyOnComputeFiles" Importance="High" />
</Target>
```

**Comment by tranb3r** [comment]
I've just realized that there was a typo in the condition (missing $) in the proposed fix.
With the correct condition, it actually works.
`<DisableFody Condition="'$(_ComputeFilesToPublishForRuntimeIdentifiers)'=='true'">true</DisableFody>`

**Comment by tranb3r** [comment]
Is it safe to keep this in my csproj?
Are you going to release a fix?
Thanks anyway!

**Comment by tom-englert** [comment]
I'd like to avoid adding a special condition for every new sdk.
Just keep this condition in your project, it's safe.

**Comment by matkoch** [comment]
Hi @tom-englert, long time no see :)

I just run into this with .NET 7 SDK, but even the mentioned condition doesn't change anything. Would it be possible to write the error under a suppressible error code that I can put into `NoWarn` ?

**Comment by matkoch** [comment]
Also, this only seems to happen when there is a second build from the same (unchanged) sources. Every time I do change something, there are no warnings.

**Comment by herebebeasties** [comment]
I think this is fixed in [6.7.0](https://www.nuget.org/packages/Fody/6.7.0) by the way.
